src/type/IntersectionType.php

```diff
@@ -18,10 +18,26 @@
 final class IntersectionType extends Type
 {
     /**
-     * @psalm-var non-empty-list<Type>
+     * @psalm-var non-empty-list<ObjectType>
      */
     private array $types;
 
+    /**
+     * @psalm-param non-empty-list<ObjectType> $types
+     */
+    public static function getFullName(Type ...$types): string
+    {
+        $names = [];
+
+        foreach ($types as $type) {
+            $names[] = $type->name;
+        }
+
+        sort($names);
```

Comment:
I really don't think this sorting is necessary here, it is an unexpected side-effect.